bin/setup

```diff
@@ -27,9 +29,40 @@ function setup::detect-bazel-version() {
   printf "%s" "${version}"
 }
 
-#——————————————————————————————  main ———————————————————————————————————————————
+function setup::header() {
```

Comment:
I wonder how much it makes sense to make setup scripts richer and richer because one of the benefit from Bazel is that your can manage dev tools themselves as a part of build configuration.

Reply:
True, but after talking to people at BazelCon it appears that shell scripts the most common "glue" between bazel and OS. 

All this script does is download pre-requisites for running bazel locally. If you see a way to shorten that great! 

I added the function to print versions of everything used mostly for CI, because it will help debugging.